src/arkhamdb/ArkhamDb.ttslua

```diff
@@ -141,29 +141,49 @@ do
     -- be changed, as later steps may act on cards added in each.  For example, a random weakness or
     -- investigator may have bonded cards or taboo entries, and should be present
     local slots = deck.slots
-    internal.maybeDrawRandomWeakness(slots, playerColor)
 
     -- handles alternative investigators (parallel, promo or revised art)
     local loadAltInvestigator = "normal"
     if loadInvestigators then
       loadAltInvestigator = internal.addInvestigatorCards(deck, slots)
     end
+    
+    internal.maybeModifyDeckFromDescription(slots, deck.description_md)
+
+    local customizations = {}
+    if deck.meta then
+      customizations = JSON.decode(deck.meta)
+    end
+
+    local deckData = {
+      slots = slots,
+      taboo_id = deck.taboo_id,
+      investigatorId = deck.investigator_code,
+      customizations = customizations,
+      playerColor = playerColor,
+      loadAltInvestigator = loadAltInvestigator
+    }
+
+    ArkhamDb.adjustDeck(deckData, callback)
+  end
 
-    internal.maybeModifyDeckFromDescription(slots, deck.description_md, playerColor)
+  ArkhamDb.adjustDeck = function(deckData, callback)
+    -- Initialize deck slot table and perform common transformations.  The order of these should not
+    -- be changed, as later steps may act on cards added in each.  For example, a random weakness or
+    -- investigator may have bonded cards or taboo entries, and should be present
+    local slots = deckData.slots
+    internal.maybeDrawRandomWeakness(slots, playerColor)
```

Comment:
Looks like `playerColor` is nil?

Reply:
Good catch!  Should be `deckData.playerColor`